src/components/modebar/buttons.js

```diff
@@ -533,8 +533,19 @@ function handleGeo(gd, ev) {
 
         if(attr === 'zoom') {
             var scale = geoLayout.projection.scale;
+            var minscale = geoLayout.projection.minscale;
+            var maxscale = geoLayout.projection.maxscale;
+
+            if(maxscale === -1) maxscale = Infinity;
             var newScale = (val === 'in') ? 2 * scale : 0.5 * scale;
 
+            // make sure the scale is within the min/max bounds
+            if(newScale > maxscale) {
```

Comment:
This line should be
```js
if(newScale >= maxscale) {
```
otherwise `newScale` would be set to `minimum` when it was equal to the `maximum`.

Reply:
Hmm. I don't get it...Looks correct to me. Only of `newScale` is out of bounce it gets forced into bounds.